src/sentry/search/events/builder/discover.py

```diff
@@ -1322,6 +1322,9 @@ def default_filter_converter(
                 1,
             )
         else:
+            # pull out the aliased expression if it exists
+            if isinstance(lhs, AliasedExpression):
```

Comment:
why do we have to do this? will this affect the other discover search uses?

Reply:
@JoshFerge it would just error out. I am pretty certain it doesn't impact discover or else we'd see the error about `AliasedExpression`

Reply:
why does discover not need this previously?

Reply:
@JoshFerge It only fails on the `project_id` input that's used as a filter. It seems like we only use this function `default_filter_converter` on small subset of filter types currently even though it's written to accommodate a lot more. I don't really know the history here and the code is convoluted. My guess is this function used to be used for more things but at some point only got used for a subset of possible filters.

Reply:
got it. mind just adding more info to that comment then?